File: ruby/averages-of-numbers.rb
# https://www.codewars.com/kata/averages-of-numbers

def averages(arr)
  averages = []
  if arr == nil
    averages
  else
  arr.each_cons(2) { |average| averages << (average.reduce(:+) / average.size.to_f) }
  averages
  end
end

File: js/averages-of-numbers.js
https://www.codewars.com/kata/averages-of-numbers/train/javascript


function averages(numbers) {
  avg = []
  if (numbers == null) {
    return avg
  } else {
    numbers.forEach(function(x, y) {
      if (y != (numbers.length - 1)) {
        avg.push((x + numbers[y + 1]) / 2)
      }
    })
    return avg
  }
}
